sql/2019/01_JavaScript/01_09.sql

```diff
@@ -0,0 +1,48 @@
+#standardSQL
+# 01_09: Changes in top JS libraries
+SELECT
+  app,
+  client,
+  freq_2018,
+  ROUND(pct_2018 * 100, 2) AS pct_2018,
+  freq_2019,
+  ROUND(pct_2019 * 100, 2) AS pct_2019,
+  ROUND((pct_2019 - pct_2018) * 100, 2) AS pct_pt_change,
+  IF(pct_2018 > 0, ROUND((pct_2019 - pct_2018) * 100 / pct_2018, 2), NULL) AS pct_change
+FROM (
+  SELECT
+    app,
+    _TABLE_SUFFIX AS client,
+    COUNT(DISTINCT url) AS freq_2018,
+    COUNT(DISTINCT url) / total AS pct_2018
+  FROM
+    (SELECT _TABLE_SUFFIX, COUNT(DISTINCT url) AS total FROM `httparchive.summary_pages.2018_07_01_*` GROUP BY _TABLE_SUFFIX)
+  JOIN
+    `httparchive.technologies.2018_07_01_*`
+  USING (_TABLE_SUFFIX)
+  GROUP BY
+    app,
+    client,
+    total)
+JOIN (
+  SELECT
+    app,
+    _TABLE_SUFFIX AS client,
+    COUNT(DISTINCT url) AS freq_2019,
+    COUNT(DISTINCT url) / total AS pct_2019
+  FROM
+    (SELECT _TABLE_SUFFIX, COUNT(DISTINCT url) AS total FROM `httparchive.summary_pages.2019_07_01_*` GROUP BY _TABLE_SUFFIX)
```

Comment:
maybe I misunderstood that convo about clients, but shouldn't the `DISTINCT url` be unnecessary here once we're grouping by client and it's a query on a pages table not requests?

Reply:
No you're right. These tables should only have one url per page. `DISTINCT` is redundant.